pennylane_lightning/src/Gates.cpp

```diff
@@ -285,6 +315,16 @@ void Pennylane::PhaseShiftGate::applyKernel(const StateVector& state, const std:
     }
 }
 
+const double Pennylane::PhaseShiftGate::generatorScalingFactor{ 1.0 };
+
+void Pennylane::PhaseShiftGate::applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices) {
+    for (const size_t& externalIndex : externalIndices) {
+        CplxType* shiftedState = state.arr + externalIndex;
+        shiftedState[indices[0]] = 0;
+        shiftedState[indices[1]] *= shift;
```

Comment:
Should this be `shift`? Since the generator is a projector onto the 1 state, maybe we just need `shiftedState[indices[0]] = 0`

Reply:
Thanks for noting this. Same here as with the 3-qubit gates. I just managed to keep the `applyDerivative` functions and forgot to update them to just apply the generator. 😞 

I'll fix them right away!